### PR TITLE
MULE-15551: Make WSC take advantage of the metadata cache.

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: src/main/kotlin/org/mule/wsdl/parser/WsdlOperationParser.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.mule.wsdl.parser.WsdlOperationTypeParser.Companion.parseInput
+import org.mule.wsdl.parser.WsdlOperationTypeParser.Companion.parseOutput
+import org.mule.wsdl.parser.model.*
+import org.mule.wsdl.parser.model.message.MessageDefinition
+import org.mule.wsdl.parser.model.operation.OperationModel
+import org.mule.wsdl.parser.model.operation.OperationType
+import org.mule.wsdl.parser.model.operation.OperationType.ONE_WAY
+import org.mule.wsdl.parser.model.operation.Type
+import org.mule.wsdl.parser.model.operation.Type.Companion.NULL_OPERATION_TYPE
+import javax.wsdl.*
```
Comment (estebanwasinger) [inline src/main/kotlin/org/mule/wsdl/parser/WsdlOperationParser.kt]: wildcards

Comment (estebanwasinger) [reply src/main/kotlin/org/mule/wsdl/parser/WsdlOperationParser.kt]: in all files

File: src/main/kotlin/org/mule/wsdl/parser/WsdlOperationTypeParser.kt [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        type.addField()
+          .key(getLocalPart(attachment))
+          .value(attachment.value)
+      }
+      return type.build()
+    }
+    return NULL_TYPE
+  }
+
+  // TODO(MULE-15275): Move this to the parser.
```
Comment (estebanwasinger) [inline src/main/kotlin/org/mule/wsdl/parser/WsdlOperationTypeParser.kt]: this is not the parser?

Comment (juandesi) [reply src/main/kotlin/org/mule/wsdl/parser/WsdlOperationTypeParser.kt]: lol

File: src/main/kotlin/org/mule/wsdl/parser/WsdlLocator.kt [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
-import javax.wsdl.xml.WSDLLocator
-
 import org.apache.cxf.wsdl11.CatalogWSDLLocator
 import org.mule.wsdl.parser.exception.WsdlParsingException
 import org.mule.wsdl.parser.locator.GlobalResourceLocator
 import org.mule.wsdl.parser.locator.ResourceLocator
 import org.xml.sax.InputSource
+import java.io.IOException
+import java.io.InputStream
+import java.util.*
```
Comment (estebanwasinger) [inline src/main/kotlin/org/mule/wsdl/parser/WsdlLocator.kt]: `*`

**Pull request merged**